MC/analysis_testing/o2dpg_analysis_test_workflow.py

```diff
@@ -149,50 +275,58 @@ def add_analysis_qc_upload_tasks(workflow, prodcution_tag, run_number, *ana_task
-            cwd = pot_ana["cwd"]
-            qc_tag = f"Analysis{ana_name_raw}"
-            needs = [ana_name]
-            for eo in expcted_outputs:
-                # this seems unnecessary but to ensure backwards compatible behaviour...
-                rename_output = eo.strip(".root")
-                rename_output = f"{rename_output}_{ana_name_raw}.root"
-                # add upload task for each expected output file
-                task = createTask(name=f"{ANALYSIS_LABEL}_finalize_{ana_name_raw}_{rename_output}", cwd=cwd, lab=[f"{ANALYSIS_LABEL}Upload", ana_name_raw], cpu=1, mem='2000', needs=needs)
-                # This has now to be renamed for upload, as soon as that is done, the output is renamed back to its original, there is in general no point of renaming it on disk only because one specific tasks needs a renamed version of it
-                rename_cmd = f"mv {eo} {rename_output}"
-                rename_back_cmd = f"mv {rename_output} {eo}"
-                task["cmd"] = f"{rename_cmd} && o2-qc-upload-root-objects --input-file ./{rename_output} --qcdb-url ccdb-test.cern.ch:8080 --task-name Analysis{ana_name_raw} --detector-code AOD --provenance qc_mc --pass-name passMC --period-name {prodcution_tag} --run-number {run_number} && {rename_back_cmd} "
-                workflow.append(task)
+        if ana_name not in analyses_to_add_for:
+            continue
+        # search through workflow stages if we can find the requested analysis
+        print(f"Adding QC upload task for analysis {ana_name_raw}")
+        pot_ana = analyses_to_add_for[ana_name]
+        cwd = pot_ana["cwd"]
+        qc_tag = f"Analysis{ana_name_raw}"
+        needs = [ana_name]
+        provenance, pass_name = ("qc_mc", "passMC") if ANALYSIS_LABEL_ON_MC in pot_ana["labels"] else ("qc_data", "pass_data")
+        for eo in ana["expected_output"]:
+            # this seems unnecessary but to ensure backwards compatible behaviour...
+            rename_output = eo.strip(".root")
+            rename_output = f"{rename_output}_{ana_name_raw}.root"
+            # add upload task for each expected output file
+            task = createTask(name=f"{ANALYSIS_LABEL}_finalize_{ana_name_raw}_{rename_output}", cwd=cwd, lab=[f"{ANALYSIS_LABEL}Upload", ana_name_raw], cpu=1, mem='2000', needs=needs)
+            # This has now to be renamed for upload, as soon as that is done, the output is renamed back to its original, there is in general no point of renaming it on disk only because one specific tasks needs a renamed version of it
+            rename_cmd = f"mv {eo} {rename_output}"
+            rename_back_cmd = f"mv {rename_output} {eo}"
+            task["cmd"] = f"{rename_cmd} && o2-qc-upload-root-objects --input-file ./{rename_output} --qcdb-url ccdb-test.cern.ch:8080 --task-name Analysis{ana_name_raw} --detector-code AOD --provenance {provenance} --pass-name {pass_name} --period-name {prodcution_tag} --run-number {run_number} && {rename_back_cmd} "
```

Comment:
```suggestion
            task["cmd"] = f"{rename_cmd} && o2-qc-upload-root-objects --input-file ./{rename_output} --qcdb-url ccdb-test.cern.ch:8080 --task-name Analysis{ana_name_raw} --detector-code AOD --provenance {provenance} --pass-name {pass_name} --period-name {production_tag} --run-number {run_number} && {rename_back_cmd} "
```
typo

Reply:
I applied that myself; there was the exact same typo in a different place which is why it worked before